### ISSUE TITLE
Error when cloning getaddrinfo() thread failed to start

### ISSUE THREAD

**Issue by bdone2012** [desktop/desktop#17755]
### The problem

Whenever I try to clone anything I get this error. It happens every time. "Cloning into 'C:\Users\Brian\Documents\GitHub\chat_app'...
fatal: unable to access 'https://github.com/bdone2012/chat_app.git/': getaddrinfo() thread failed to start"

I wanted to clone Stable Diffusion but it wasn't working so I tried an old sample app of my own and that wouldn't even work.

Note this computer has never had git or github on it before. I normally use git on mac, but now it's not working on windows the way I'd expect. I tried fresh installations of git.

Let me know if any other info would be helpful. Thanks

### Release version

Version 3.3.5 (x64

### Operating system

Windows 10

### Steps to reproduce the behavior

Simply trying to clone anything.

### Screenshots

![image](https://github.com/desktop/desktop/assets/6579647/c9215297-bccb-4309-942c-dda157da784e)

**Comment by tidy-dev** [comment]
@bdone2012 Sorry for your troubles. This looks like you may be having similar troubles to the user in another ticket. [Here is what they posted worked for them.](https://github.com/desktop/desktop/issues/17069#issuecomment-1636513954) . Could you give that a try and see if helps?

Other thoughts are if there could be any firewall or antivirus preventing access to github.

**Comment by bdone2012** [comment]
@tidy-dev Thanks so much for your help. I'm still having issues. I had already seen the above issue that you linked but I tried it again to make sure I did it properly and it still didn't work. I also tried the following:

`git config --system http.sslcainfo 'C:\Program Files\Git\mingw64\etc\ssl\certs\ca-bundle.crt'`

 Because my Git is 64 not 32 so the file is in a different spot. I did also try removing the 64 git version and downloaded the 32 one, although I'm not positive I did that after running: 

`git config --system http.sslcainfo 'C:\Program Files (x86)\Git\mingw32\ssl\certs\ca-bundle.crt'`

Would the 32 git be better than the 64?

I turned off the 3 firewall settings that are in Firewall & Network Protections and it didn't help. I don't have other firewalls as far as I can remember.

Now I'm wondering if there is something wrong with my ca-bundle.crt because the person's issue that you linked to really does seem like the exact problem I'm getting. So even if their specific fix didn't work maybe the issue is in the same area. 

I'm looking at this thread and people are talking about removing a line but I haven't found it yet so I'm not sure their exact fix will work either. Thanks again

https://github.com/desktop/desktop/issues/4832

**Comment by tidy-dev** [comment]
Good investigation. I was zeroing in on the more unique part of your error. `getaddrinfo() thread failed to start`. It may be helpful to show the contents of that mentioned config output to see if I can pin point anything to take a look at. Mind sharing it?

```
git config -l --show-origin
```

**Comment by bdone2012** [comment]
Here it is:

$ git config -l --show-origin

file:C:/Program Files/Git/etc/gitconfig diff.astextplain.textconv=astextplain
file:C:/Program Files/Git/etc/gitconfig filter.lfs.clean=git-lfs clean -- %f
file:C:/Program Files/Git/etc/gitconfig filter.lfs.smudge=git-lfs smudge -- %f
file:C:/Program Files/Git/etc/gitconfig filter.lfs.process=git-lfs filter-process
file:C:/Program Files/Git/etc/gitconfig filter.lfs.required=true
file:C:/Program Files/Git/etc/gitconfig http.sslbackend=openssl
file:C:/Program Files/Git/etc/gitconfig http.sslcainfo=C:\Program Files\Git\mingw64\etc\ssl\certs\ca-bundle.crt
file:C:/Program Files/Git/etc/gitconfig http.sslverify=false
file:C:/Program Files/Git/etc/gitconfig core.autocrlf=true
file:C:/Program Files/Git/etc/gitconfig core.fscache=true
file:C:/Program Files/Git/etc/gitconfig core.symlinks=false
file:C:/Program Files/Git/etc/gitconfig core.editor="C:\\Program Files\\Sublime Text 3\\subl.exe" -w
file:C:/Program Files/Git/etc/gitconfig pull.rebase=false
file:C:/Program Files/Git/etc/gitconfig credential.https://dev.azure.com.usehttppath=true
file:C:/Program Files/Git/etc/gitconfig init.defaultbranch=master
file:C:/Users/Brian/.gitconfig  filter.lfs.required=true
file:C:/Users/Brian/.gitconfig  filter.lfs.clean=git-lfs clean -- %f
file:C:/Users/Brian/.gitconfig  filter.lfs.smudge=git-lfs smudge -- %f
file:C:/Users/Brian/.gitconfig  filter.lfs.process=git-lfs filter-process
file:C:/Users/Brian/.gitconfig  user.name=bdone2012
file:C:/Users/Brian/.gitconfig  user.email=bdone2012@gmail.com
file:C:/Users/Brian/.gitconfig  https.sslverify=false
file:.git/config        core.repositoryformatversion=0
file:.git/config        core.filemode=false
file:.git/config        core.bare=false
file:.git/config        core.logallrefupdates=true
file:.git/config        core.symlinks=false
file:.git/config        core.ignorecase=true

Thanks!

**Comment by bdone2012** [comment]
Oh also my anti virus was on but after you mentioned it I turned it all off. Then I turned it back on and followed these instructions to allow exceptions for both the github desktop app folder and for the git folder so I don't think this is the problem. But I don't know much about windows anti virus stuff.

https://github.com/stascorp/rdpwrap/issues/1112

Also a final note I have Windows Home. Although I wouldn't think that'd make a difference. Thanks

**Comment by tidy-dev** [comment]
Hmm.. That config looks nearly identical to mine.. so I don't suspect anything there.

The error is emanating from git (not specific to GitHub Destktop). Thus, I did some stack overflow searching and came across https://stackoverflow.com/questions/59911649/fatal-unable-to-access-link-getaddrinfo-thread-failed-to-start

Seems the most common answer is firewall settings -> some point to various windows defender firewall. Can you try the suggestions int that stack overflow?

**Comment by bdone2012** [comment]
I tried disabling it as they said in that post and the github app still won't let me clone. Someone suggests deleting the firewall completely but on another post on stackoverflow that describes how to do it they warn very strongly against deleting it completely. Sounds like the only way to get it back after that is to factory reset the computer. I would do it if necessary but that seems like a last resort

![image](https://github.com/desktop/desktop/assets/6579647/643833c4-e3df-4726-aa15-d3605b24e440)

**Comment by bdone2012** [comment]
This is where they talk about deleting it and they warn this shouldn't be used on a computer that is functioning
https://superuser.com/questions/1390124/is-it-possible-to-remove-defender-antivirus-and-firewall-from-the-latest-window

**Comment by tidy-dev** [comment]
Yeah. And many of our users have Windows with the Windows Defender firewall active with no issue (it is active on my Windows VM), so I can't imagine you would have to completely remove it. My thinking that maybe some setting with it needs adjusted.  But.. I am sorry to say I do not know a lot about the individual settings and can't really provide anymore input then those threads.

Last ditch effort, did you try restarting the pc after turning it off?

**Comment by steveward** [comment]
@bdone2012 I'm going to close this out as we have not yet heard back from you, but let us know if this is still an issue and we can investigate further.

**Comment by AntAgna** [comment]
I have had the same issue. "getaddrinfo() thread failed to start" error message for all repos when trying to fetch or pull from any URL. DNS resolution seemed to work fine. The problem stopped after rebooting Windows.

**Comment by singlecellaa** [comment]
I tried switching from https to ssh for my git repository, and it finally worked

**Comment by hpst3r** [comment]
Just a note. Posted here because this is one of the top search results for "getaddrinfo() thread failed to start" and it might be helpful? Maybe?

This could be (at least sometimes) a Windows socket error. Certainly not specific to Github Desktop. I believe this is a curl error message.

On my system, Git Bash's default curl is `/mingw64/bin/curl`, which is:

```
curl 8.10.1 (x86_64-w64-mingw32) libcurl/8.10.1 Schannel zlib/1.3.1 brotli/1.1.0 zstd/1.5.6 libidn2/2.3.7 libpsl/0.21.5 libssh2/1.11.0
Release-Date: 2024-09-18
```

I have not seen this occur anywhere aside from Git. I don't think I've seen it occur when using SSH (only HTTPS).

All of my systems are running W11 24H2 (26100) - I see this error on occasion while I am connected to VPN (I believe it has only happened to me with both the Tailscale client when it is mucking with DNS responses, but am not 100%) and I am attempting to perform a HTTP network operation (typically push/pull/fetch) with Git, typically to/from Github.

My machines are running SentinelOne OR Windows Defender. Firewalls are typically default, sometimes they'll have a few incoming exceptions.

Disconnecting from VPN resolves this every time I've had it reoccur. It only occurs intermittently, and I can reconnect, then continue making requests immediately afterwards without it being a problem again for days or weeks at a time.

I believe I have experienced this on three machines (not 100%), all 24H2, with roughly the same set of software installed, including Tailscale, which is my AOVPN.

The specific error message I am seeing:

```
fatal: unable to access 'https://github.com/hpst3r/oneofmyrepos.git/': getaddrinfo() thread failed to start
```

Anyway, this is the second or third time I've come back to this post, so I figured I'd add a comment. Apologies if this is wildly off-topic.
